enhancements/network/ovn-kubernetes-ipsec.md

```diff
@@ -0,0 +1,298 @@
+  certificate.
+- Modify OVNKubernetes operator configuration object in order to allow
+  enablement of IPsec for the cluster.
+- Certificate rotation will be handled in the same as it is with the current
+  OVN TLS implementation.
+
+### Risks and Mitigations
+
+#### Upstreaming
+
+A number of changes will need to be made to upstream communities including the
+Linux kernel. As these may need time to execute, it is imperative that we
+identify them early and start the process.
+
+#### Performance
+
+Enablement of IPsec will cause degradation of overall networking throughput and
+latency due to the additional overhead of the IPsec protocol. Some early
+performance testing will be done in a non-production cluster in order to get
+an approximation of the expected performance degradation - These results will be
+posted at a later date. After implementation of the enhancement, it is expected
+that comprehensive benchmarking and profiling will take place. Based on the
+output of this activity, it may be decided to start some optimization activities
+in parallel to improve performance.
+
+#### Security Review
+
+See "Open Questions" below.
+
+## Design Details
+
+### Open Questions [optional]
+
+- Will this need to go through some kind of Security Review?
+
+### Test Plan
+
+It will expected to add a number of Continuous Integration jobs for IPsec:
+
+- All Openshift e2e tests should be replicated in a cluster that has been
+  installed with IPsec enabled. This should ensure IPsec has no regressions with
+  respect to a cluster that has not been enabled with IPsec.
+- In order to test any issues with version skew, an IPsec-enabled cluster should
+  be upgraded while active.
+
+Additional CI jobs could potentially be added at a later stage.
+
+- Functional tests
+    - Ensure certificate expiration and rotation happens correctly.
+- Non-functional tests
+    - Performance tests (scale tests)
+    - Endurance tests (traffic generation over a long period)
+    - Recovery tests (what happens when node IPsec components such as pluto or
+      ovs-monitor-ipsec are killed)
+
+It should also be noted that, some upstream OVS/OVN unit and integration tests
+will also need to be developed.
+
+### Graduation Criteria
+
+TBD
+
+#### Examples
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Performance measurement
+
+##### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+##### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+Currently, it is proposed that this feature can only be enabled at cluster
+installation time.
+
+### Version Skew Strategy
+
+It is proposed that this enhancement will not support mixed clusters of RHEL7
+and RHEL8 nodes. This can be supported in a later enhancement.
```

Comment:
This section is just about OCP versions... Not supporting mixed RHEL 7 and RHEL 8 belongs in "Non-Goals". (And I assume what you really mean is "we won't support RHEL 7", right?)

Reply:
Yes, you are right. I will move it to goals. I don't think I am saying that we won't support RHEL7. However, based on your comment, you must have had a thought about this? Can you foresee some difficulty?

Reply:
I was just assuming that if we are depending on patches that aren't yet in the RHEL 8 kernel, then the RHEL 7 kernel must be even less usable...

Reply:
Ok, thats clear now.